app/src/utils/spatial-utils.tsx

```diff
@@ -75,7 +75,7 @@ export const parseSpatialDataByType = (
         if (visible) {
           boundaryStaticLayer.features.push({
             geoJSON: feature,
-            key: feature.id || feature.properties.id,
+            key: feature.id || feature.properties.datasetID,
```

Comment:
This was undefined before, and so the feature key was defaulting to a generic index, which isn't unique, and was causing funny behaviour when toggling the marker visibility.